### PR TITLE
Add KafkaSource migration logic as a post-install job

### PR DESCRIPTION
**Pull request by pierDipi**
The existing KafkaSource creates a deployment for each instance
and nothing more.

This is what it's in the namespace where I created a `KafkaSouce`.
Only the
`deployment.apps/kafkasource-kafka-source-2e0b5e79-8547-46e8-b018-aac56c6c8d16`
is created automatically by the
`kafka-controller-manager`, other resources are sink services
that shouldn't be deleted during the migration.
```
pierdipi@pierdipi eventing-kafka-broker (kafka-source-migration-post-install) $ k get all
NAME                                                                  READY   STATUS    RESTARTS   AGE
pod/kafkasource-kafka-source-2e0b5e79-8547-46e8-b018-aac56c6c8bhp5p   1/1     Running   0          106m
pod/label-demo                                                        1/1     Running   0          106m
pod/label-demo-2                                                      1/1     Running   0          106m

NAME                      TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)   AGE
service/event-display     ClusterIP   10.96.16.19    <none>        80/TCP    106m
service/event-display-2   ClusterIP   10.96.177.33   <none>        80/TCP    106m
service/kubernetes        ClusterIP   10.96.0.1      <none>        443/TCP   116m

NAME                                                                            READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/kafkasource-kafka-source-2e0b5e79-8547-46e8-b018-aac56c6c8d16   1/1     1            1           106m

NAME                                                                                       DESIRED   CURRENT   READY   AGE
replicaset.apps/kafkasource-kafka-source-2e0b5e79-8547-46e8-b018-aac56c6c8d16-695fbb68b8   1         1         1       106m

NAME                                           TOPICS                   BOOTSTRAPSERVERS                            READY   REASON   AGE
kafkasource.sources.knative.dev/kafka-source   ["knative-demo-topic"]   ["my-cluster-kafka-bootstrap.kafka:9092"]   True             106m
```

The plan for post-install logic is to have a generic post-install
job in eventing-kafka-broker (in addition to a storage version
migrator) this is the job where we place general post-install
operations (manifests placed in control-plane/config/post-install).

In addition I moved the manifests for the control plane to
`control-plane/config/eventing-kafka-broker` to easily produce
the `eventing-kafka-broker.yaml` manifest via recursive
resolution.

*Testing*

- Apply `source.yaml` from `eventing-kafka`.
- Create `KafkaSouce`(s)
- Scale down the `kafka-controller-manager` deployment with:
  `k scale deployment -n knative-eventing kafka-controller-manager --replicas=0`
- Create the post-install manifest with:
  `ko resolve -Rf control-plane/config/post-install/ > eventing-kafka-post-install.yaml`
- Run the post-install job with:
  `ko create -f eventing-kafka-post-install.yaml`

Output:
```
pierdipi@pierdipi eventing-kafka-broker (kafka-source-migration-post-install) $ k get all
NAME               READY   STATUS    RESTARTS   AGE
pod/label-demo     1/1     Running   0          124m
pod/label-demo-2   1/1     Running   0          124m

NAME                      TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)   AGE
service/event-display     ClusterIP   10.96.16.19    <none>        80/TCP    124m
service/event-display-2   ClusterIP   10.96.177.33   <none>        80/TCP    124m
service/kubernetes        ClusterIP   10.96.0.1      <none>        443/TCP   134m

NAME                                           TOPICS                   BOOTSTRAPSERVERS                            READY   REASON   AGE
kafkasource.sources.knative.dev/kafka-source   ["knative-demo-topic"]   ["my-cluster-kafka-bootstrap.kafka:9092"]   True             124m
```

Signed-off-by: Pierangelo Di Pilato <pierdipi@redhat.com>

## Proposed Changes

- Add KafkaSource migration logic as a post-install job

**Release Note**

```release-note
Add KafkaSource migration logic as a post-install job (`eventing-kafka-post-install.yaml`).
```

Next: left-over global resources deletion

### CODE REVIEW

File: control-plane/config/post-install/200-controller-cluster-role.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      - delete
+      - deletecollection
+
+  # sources.knative.dev resources and finalizers we care about.
+  - apiGroups:
+      - sources.knative.dev
+    resources:
+      - kafkasources
+      - kafkasources/status
+    verbs:
```
Comment (matzew) [inline control-plane/config/post-install/200-controller-cluster-role.yaml:38]: are these verbs enough?

File: control-plane/config/post-install/200-controller-cluster-role.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+rules:
+  - apiGroups:
+      - apps
+    resources:
+      - deployments
+    verbs:
+      - delete
+      - deletecollection
+
+  # sources.knative.dev resources and finalizers we care about.
```
[the remark this answers is not in the document]
Comment (pierDipi) [reply control-plane/config/post-install/200-controller-cluster-role.yaml:32]: No need for finalizers, I copy-pasted the text, I'll remove this comment.

File: control-plane/config/post-install/200-controller-cluster-role.yaml (hunk 1 above)
Comment (pierDipi) [reply control-plane/config/post-install/200-controller-cluster-role.yaml:38]: Yes, these are enough

- list Kafka sources: https://github.com/knative-sandbox/eventing-kafka-broker/pull/1889/files#diff-736787d0a0ff25859f1fa32f6413a911418e4676d53deddb52980359818eaa24R42-R45
- and then delete collection: https://github.com/knative-sandbox/eventing-kafka-broker/pull/1889/files#diff-736787d0a0ff25859f1fa32f6413a911418e4676d53deddb52980359818eaa24R64-R67

**Pull request merged**